src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/LoansServiceImpl.java

```diff
@@ -74,15 +78,24 @@
 
         LoanApi loanApi = loanTransformer.getLoanApi(loanToAdd);
 
+        List<ValidationError> validationErrors = loanValidator.validateDirectorToAdd(loanToAdd);
+
+        if(!validationErrors.isEmpty()) {
```

Comment:
If we return before we make any API calls, we could move the logic above this line, further down? (ApiClient creation and possibly the loan transform)